src/pages/registration/contributors_tab/Contributors.tsx

```diff
@@ -0,0 +1,178 @@
+  const dispatch = useDispatch();
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<Registration>();
+  const {
+    entityDescription: { contributors },
+  } = values;
+  const [openContributorModal, setOpenContributorModal] = useState(false);
+  const [unverifiedContributor, setUnverifiedContributor] = useState<UnverifiedContributor | null>(null);
+  const isMobile = useIsMobile();
+
+  const relevantContributors = contributors.filter((contributor) => contributor.role === contributorRole);
+  const otherContributors = contributors.filter((contributor) => contributor.role !== contributorRole);
+
+  const handleOnRemove = (indexToRemove: number) => {
+    const nextRelevantContributors = relevantContributors
+      .filter((_, index) => index !== indexToRemove)
+      .map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+    const nextContributors = [...nextRelevantContributors, ...otherContributors];
+    setFieldValue(ContributorFieldNames.CONTRIBUTORS, nextContributors);
+
+    if (nextContributors.length === 0) {
+      // Ensure field is set to touched even if it's empty
+      setFieldTouched(ContributorFieldNames.CONTRIBUTORS);
+    }
+  };
+
+  const handleMoveContributor = (newSequence: number, oldSequence: number) => {
+    const oldIndex = relevantContributors.findIndex((c) => c.sequence === oldSequence);
+    const minNewIndex = 0;
+    const maxNewIndex = relevantContributors.length - 1;
+
+    const boundedNewIndex =
+      newSequence > relevantContributors.length ? maxNewIndex : newSequence < 0 ? minNewIndex : null;
+    const newIndex = boundedNewIndex
+      ? boundedNewIndex
+      : Math.min(
+          maxNewIndex,
+          Math.max(
+            minNewIndex,
+            relevantContributors.findIndex((c) => c.sequence === newSequence)
+          )
+        );
+
+    const reorderedContributors = move(relevantContributors, oldIndex, newIndex) as Contributor[];
+    // Ensure incrementing sequence values
+    const newContributors = reorderedContributors.map((contributor, index) => ({
+      ...contributor,
+      sequence: index + 1,
+    }));
+    setFieldValue(ContributorFieldNames.CONTRIBUTORS, [...otherContributors, ...newContributors]);
+  };
+
+  const handleOpenContributorModal = (unverifiedAuthor: UnverifiedContributor) => {
+    setUnverifiedContributor(unverifiedAuthor);
+    setOpenContributorModal(true);
+  };
+
+  const onAuthorSelected = (authority: Authority) => {
+    if (relevantContributors.some((author) => author.identity.id === authority.id)) {
+      dispatch(setNotification(t('contributors.author_already_added'), NotificationVariant.Info));
+      return;
+    }
+
+    const identity: Identity = {
+      type: BackendTypeNames.IDENTITY,
+      id: authority.id,
+      orcId: authority.orcids.length > 0 ? authority.orcids[0] : '',
+      name: authority.name,
+    };
+
+    if (!unverifiedContributor) {
+      const newAuthor: Contributor = {
```

Comment:
Bruke `contributor` som en samlebetegnelse på alle typer som kan være her? også er author en type contributor liksom?
```suggestion
      const newContributor: Contributor = {
```